include/power/rt_dpm.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2020 Intel corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_POWER_RT_DPM_H
+#define ZEPHYR_INCLUDE_POWER_RT_DPM_H
+
+#include <device.h>
+#include <sys/atomic.h>
+#include <kernel_structs.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief States defined for device runtime power management.
+ */
+enum rt_dpm_state {
+	RT_DPM_ACTIVE,
+	RT_DPM_RESUMING,
+	RT_DPM_SUSPENDED,
+	RT_DPM_SUSPENDING
+};
+
+/**
+ * @brief Callbacks defined for device runtime power management.
+ *
+ * The resume_prepare/suspend_prepare callbacks are used to do things
+ * without irq locked (e.g. states sync between driver and firmware).
+ * Device driver should take care all the possible errors during this
+ * irq unlocked period and the result will be indicated by the return
+ * value.
+ *
+ * The resume/suspend callbacks will do the real power/clock involved
+ * things with irq locked.
+ *
+ * @retval 0 if successfully excute resume_prepare/suspend_prepare.
+ * @retval -EIO if error happens during the prepare stage.
+ */
+struct rt_dpm_ops{
+	void (*resume)(struct device *dev);
```

Comment:
> Instead of providing resume/suspend callbacks, would it not be simpler to just call into device_set_power_state() for better integration with the rest of the device power management framework, and save a few bytes in the process?

Will consider it, thanks. 

> 
> Also, "resume_prepare" is a bit ambiguous. Looks like it has nothing to do with preparing to resume, since "resume" is called before "resume_prepare". Personally I'd rather give them names such as "pre_suspend" and "post_resume".

Thanks